packages/main/src/TextArea.ts

```diff
@@ -0,0 +1,710 @@
+	 * when <code>name</code> property is set.
+	 * @type {HTMLElement[]}
+	 * @name sa.ui.webc.main.TextArea.prototype.formSupport
+	 * @slot
+	 * @private
+	 */
+	 @slot()
+	 formSupport!: Array<HTMLElement>;
+
+	_fnOnResize: ResizeObserverCallback;
+	_firstRendering: boolean;
+	_openValueStateMsgPopover: boolean;
+	_exceededTextProps!: ExceededText;
+	_keyDown?: boolean;
+	FormSupport?: typeof FormSupportT;
+	previousValue: string;
+	popover?:Popover;
+
+	static get styles() {
+		return [browserScrollbarCSS, styles];
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get template() {
+		return TextAreaTemplate;
+	}
+
+	static get staticAreaTemplate() {
+		return TextAreaPopoverTemplate;
+	}
+
+	static get staticAreaStyles() {
+		return valueStateMessageStyles;
+	}
+
+	static i18nBundle: I18nBundle;
+
+	static async onDefine() {
+		TextArea.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	constructor() {
+		super();
+
+		this._firstRendering = true;
+		this._openValueStateMsgPopover = false;
+		this._fnOnResize = this._onResize.bind(this);
+		this.previousValue = "";
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._fnOnResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._fnOnResize);
+	}
+
+	onBeforeRendering() {
+		this._exceededTextProps = this._calcExceededText();
+		this._mirrorText = this._tokenizeText(this.value);
+
+		this.exceeding = !!this._exceededTextProps.leftCharactersCount && this._exceededTextProps.leftCharactersCount < 0;
+		this._setCSSParams();
+
+		const FormSupport = getFeature<typeof FormSupportT>("FormSupport");
+		if (FormSupport) {
+			FormSupport.syncNativeHiddenInput(this);
+		} else if (this.name) {
+			console.warn(`In order for the "name" property to have effect, you should also: import "@ui5/webcomponents/dist/features/InputElementsFormSupport.js";`); // eslint-disable-line
+		}
+	}
+
+	onAfterRendering() {
+		const nativeTextArea = this.getInputDomRef()!;
+
+		if (this.rows === 1) {
+			nativeTextArea.setAttribute("rows", "1");
+		} else {
+			nativeTextArea.removeAttribute("rows");
+		}
+
+		this.toggleValueStateMessage(this.openValueStateMsgPopover);
+		this._firstRendering = false;
+	}
+
+	getInputDomRef() {
+		return this.getDomRef()!.querySelector<HTMLTextAreaElement>("textarea")!;
+	}
+
+	_onkeydown(e: KeyboardEvent) {
+		this._keyDown = true;
+
+		if (isEscape(e)) {
+			const nativeTextArea = this.getInputDomRef();
+
+			this.value = this.previousValue;
+			nativeTextArea.value = this.value;
+			this.fireEvent("input");
+		}
+	}
+
+	_onkeyup() {
+		this._keyDown = false;
+	}
+
+	_onfocusin() {
+		this.focused = true;
+		this._openValueStateMsgPopover = true;
+		this.previousValue = this.getInputDomRef().value;
+	}
+
+	_onfocusout(e: FocusEvent) {
+		const eTarget = e.relatedTarget as HTMLElement;
+		const focusedOutToValueStateMessage = eTarget?.querySelector(".ui5-valuestatemessage-root");
```

Comment:
Previously you were looking for element inside this element's shadow DOM. Could you recheck if this change is okey?

`event.relatedTarget && event.relatedTarget.shadowRoot && event.relatedTarget.shadowRoot`